02 Toolbar/src/common/managers/textAreaManager.ts

```diff
@@ -0,0 +1,60 @@
+
+// Based on: http://stackoverflow.com/questions/1064089/inserting-a-text-where-cursor-is-using-javascript-jquery
+class TextAreaManager {
+  insertAtCaret(textArea: HTMLTextAreaElement, caret: string, offsetCursor: number = 0): string {
```

Comment:
Inside this method I'm using the private `buildTextWithCaretBetweenSelectedText` but I think it is too large. Maybe `buildTextWithCaret`?